### PR TITLE
task: configure alarm on CampaignSendMessagePermanentFailure metric

### LINKED ISSUE #34: Account requires additional review prior to being able to send SMS messages
I finally got around to trying to send me an SMS. I didn't receive one and saw no errors in the logs or CloudTrail. After further investigation Pinpoint campaign details show that one SMS was sent but none were delivered, resulting in a delivery rate of 0%.

Only when I tried to send myself another message via pinpoint's test feature, it showed me this:

> Message not sent: We weren't able to send your SMS message to +49172xxxxxxx.
> Details
> Request Id 415 : Your account requires additional review prior to being able to send SMS messages. Please create a support ticket with an explanation of your use case. https://console.aws.amazon.com/support/cases#/create?issueType=service-limit-increase&limitType=service-code-sns-text-messaging

I reckon you went through that process @cremich ? Should this maybe be noted in the Readme?

More importantly, alerting on the delivery rate of campaigns would be good to have. Failure to deliver the message, moreover silently, undermines the whole idea.

**To Reproduce**
Use an AWS account that was never used/unlocked for SMS sending before.

### PR DESCRIPTION
**Pull request by Bakkenrak**
# SMS delivery failure alarming

## Description

Configure alarm on CampaignSendMessagePermanentFailure metric which is triggered when an SMS delivery fails. Comes with a stack for shared alarm infrastructure (i.e. SNS topic with email subscriptions).

Fixes #34 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Test deployments and unit tests.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by cremich** [comment]
Awesome work @Bakkenrak . This is really helpful!

**Review by cremich — APPROVED** [review]

File: bin/kiko.ts [hunk 1]
```diff
@@ -5,20 +5,27 @@ import { PoolManagementStack } from "../lib/pool-management/stack";
...
 import { AmplifyStack } from "../lib/amplify/stack";
+import { AlarmInfra } from "../lib/shared/constructs/alarm-infra";
 
 const app = new cdk.App();
 const tags = app.node.tryGetContext("tags");
 const deployStage = process.env.DEPLOY_STAGE || "dev";
 cdk.Tags.of(app).add("project", tags.project);
 cdk.Tags.of(app).add("stage", deployStage);
 
+const alarmInfra = new AlarmInfra(app, `${deployStage}-kiko-alarm-infra`, {
```
[the remark this answers is not in the document]
Comment (Bakkenrak) [reply bin/kiko.ts:16]: yes, good point

File: bin/kiko.ts [hunk 2]
```diff
@@ -5,20 +5,27 @@ import { PoolManagementStack } from "../lib/pool-management/stack";
...
+  deployStage,
+  commaSeperatedEmailAddresses: process.env.ALARM_SUBSCRIPTION_EMAIL_ADDRESSES,
```
[the remark this answers is not in the document]
Comment (Bakkenrak) [reply bin/kiko.ts:18]: I was kinda orienting on the deploy stage which is also passed as an env var. But the context is a better idea indeed.

If no email address is supplied, no subscription to the alarm topic will be made.

File: lib/tenant-management/constructs/tenant.ts [hunk 3]
```diff
@@ -36,11 +40,32 @@ export class Tenant extends cdk.Construct {
...
     });
 
+    const smsSendFailureMetric = new cloudwatch.Metric({
+      namespace: "AWS/Pinpoint",
+      metricName: "CampaignSendMessagePermanentFailure",
+      period: Duration.minutes(1),
+      statistic: "sum",
+      dimensions: {
+        Channel: "SMS",
+        ApplicationId: smsChannel.applicationId,
```
[the remark this answers is not in the document]
Comment (Bakkenrak) [reply lib/tenant-management/constructs/tenant.ts:55]: Actually `this.pinpointApplication.ref` might! It didn't look appropriate from what I interpreted from the documentation but it propably is.

**Pull request merged**